### DISCUSSION TITLE
How to derive the unit of the product of two quantities?

### DISCUSSION THREAD

**Discussion by giacomociti** [qudt/qudt-public-repo#1044] [Q&A]
Maybe I'm missing something fundamental here, but I expected a way to derive from, say, 2 m/sec and 3 sec, that their product is 6m.
Using dimension vectors I can derive the quantity kind of the product, but there are many applicable units

**Comment by jhodgesatmb** [comment]
This is a very common/ubiquitous operation performed on most any equation in the physical sciences, but it is an extension of QUDT. We have developed a few applications of this kind over time but they have all been experiments rather than being part of the release.Jack Hodges, Ph.D.Arbor StudiosOn Nov 26, 2024, at 6:38 AM, Giacomo Citi ***@***.***> wrote:﻿
Maybe I'm missing something fundamental here, but I expected a way to derive from, say, 2 m/sec and 3 sec, that their product is 6m.
Using dimension vectors I can derive the quantity kind of the product, but there are many applicable units

**Comment by fkleedorfer** [comment]
You can do these kinds of calculations using 

- https://github.com/occamsystems/occam-qudt
- https://github.com/qudtlib/qudtlib-java
- https://github.com/qudtlib/qudtlib-js

However, not each one of them is equally suited. For what you are looking for, try the first one first.

**Comment by Rdataflow** [comment]
you may simply go with a query adding up and filtering for the matching dimension vector
i.e. https://qlever.cs.uni-freiburg.de/lindas/6wpQVJ
btw: Newton meter = Joule = Watt seconds

**Comment by giacomociti** [comment]
thanks for the answers and the pointers. I was looking for a SPARQL solution, so I built upon @Rdataflow suggestion and this is the best I could do: https://qlever.cs.uni-freiburg.de/lindas/jUdDLz

**Comment by Rdataflow** [comment]
@giacomociti it feels a bit hacky but I can propose a possible solution:
https://qlever.cs.uni-freiburg.de/lindas/eTQvRt 😎

**Comment by VladimirAlexiev** [comment]
slightly related:
- https://github.com/qudt/qudt-public-repo/issues/1207
